File: search/jump_search/php/jump_search.php
<?php

function jumpSearch(array $elements, $number) {
    $n = count($elements);
    
    // Finding block size to be jumped
    $step = sqrt($n);
    
    // Finding the block where element is
    // present (if it is present)
    $prev = 0;
    while($elements[min($step,$n)-1] < $number)
    {
        $prev = $step;
    	$step += sqrt($n);
    	if($prev >= $n)
        	return -1;
    }
    
    // Doing a linear search for x in block
    // beginning with prev.
    while($elements[$prev] < $number)
    {
        $prev++;
        
        // If we reached next block or end of
        // array, element is not present.
        if ($prev === min($step, $n))
            return -1;
    }
    
    // If element is found
    if ($elements[$prev] === $number)
        return $prev;
    
    return -1;
}
function jumpSearchTest() {
    $elements = [1,2,3,5,21,42,63,256,1024];
    $num = 63;
    $result = jumpSearch($elements, $num);
    
    if ($result === -1) {
        echo "Element not found.";
    } else {
        echo "Element found.";
    }
}
jumpSearchTest();

?>

File: search/jump_search/C#/JumpSearch.cs
using System;

public class JumpSearch
{
    // Testing the program.
    public static void Main(string[] args)
    {
        int[] arr = { 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610 };
        int x = 55;
        int y = 14;
        Console.WriteLine(
            $"{JumpSearch(arr, x)}\n{JumpSearch(arr, y)}");
    }

    public static int JumpSearch(int[] array, int number)
    {
        Array.Sort(array);
        int n = array.Length;
        // Finding a block size to be jumped.
        int block = (int)(Math.Floor(Math.Sqrt(n)));
        int prev = 0;

        // Finding an index at which number is less than element at index.
        while (array[Math.Min(block, n) - 1] < number)
        {
            prev = block;
            block += block;
            if (prev >= n)
            {
                return -1;
            }
        }

        // Linear search for number in block beginning with prev.
        while (array[prev] < number)
        {
            prev++;
            if (prev == Math.Min(block, n))
            {
                return -1;
            }
        }

        return array[prev] == number ? prev : -1;
    }
}